sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,157 @@
+            mask = self.mask[item]
+        else:
+            mask = None
+        if data.ndim is 2:
+            result = Cube2D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        elif data.ndim is 1:
+            result = Cube1D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        else:
+            result = NDCube(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        return result
+
+
+class Cube2D(NDCube):
+    """docstring for Cube2D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube2D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self, axes=None, axis_data=['x', 'y'], **kwargs):
+        """
+        Plots an x-y graph at a certain specified wavelength onto the current
+        axes. Keyword arguments are passed on to matplotlib.
+
+        Parameters
+        ----------
+        axes: `astropy.visualization.wcsaxes.core.WCSAxes` or `None`:
+            The axes to plot onto. If None the current axes will be used.
+
+        axis_data: `list`.
+            The first axis in WCS object will become the first axis of axis_data and
+            second axis in WCS object will become the seconf axis of axis_data.
+        """
+        if axes is None:
+            if self.wcs.naxis is not 2:
+                slice_list = self.wcs._naxis
+                axis_index = []
+                for i, ax in enumerate(slice_list):
+                    if ax is not 1:
+                        axis_index.append(i)
+                if len(axis_index) is 2:
+                    slice_list[axis_index[0]] = axis_data[0]
+                    slice_list[axis_index[1]] = axis_data[1]
+                else:
+                    raise ValueError("Dimensions of WCS and data don't match")
+            axes = wcsaxes_compat.gca_wcs(self.wcs, slices=slice_list)
+        plot = axes.imshow(self.data, **kwargs)
+        return plot
+
+
+class Cube1D(NDCube):
+    """docstring for Cube1D"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        super(Cube1D, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def plot(self, unit=None):
+        """
+        Plots a graph.
+        Keyword arguments are passed on to matplotlib.
+
+        Parameters
+        ----------
+        unit: `astropy.unit.Unit`
+        The data is changed to the unit given or the self.unit if not given.
+        """
+        if self.unit is not None and unit is None:
```

Comment:
The x-axis values should be derived here from the WCS object as a Quantity.  It should then be used in the subsequent plot commands, e.g. ```plot = plt.plot(x_axis, self.data*self.unit)```

There should probably also be an ```xunit=None``` kwarg to allow the user to set the unit of the xaxis.  Something like:
```
axis_ind = 0
x_axis = Quantity(self.pixel_to_world(np.arange(self.wcs._naxis[axis_ind]), unit=self.wcs.wcs.cunit[axis_ind])
if isinstance(xunit, u.Unit) or isinstance(xunit, str):
   x_axis = x_axis.to(xunit)
```
You will probably have to verify the value of ```axis_ind``` in a similar way to what you did in Cube2D plot, i.e. in the case where there are more WCS axes than data axes because of inseparable dimensions.  I haven't tested the above code.  So it's just to give you an idea of what I mean.